examples/deal.II/README.MD

```diff
@@ -0,0 +1,20 @@
+An example how to write libCEED operators (BP1-BP6) within the open-source
+finite element library [deal.II](https://www.dealii.org/). As reference,
+operators are presented that use the native matrix-free infrastructure.
+
+First compile deal.II and libCEED individually. After that, compile the deal.II example:
+
+```bash
+mkdir build
+cd build
+cmake ../ -DDEAL_II_DIR=~/path/to/dealii -DCEED_DIR=~/path/to/libceed
+make
+```
+
+To run the executable, write:
+
+```
+./bp
+```
+
+Optional command-line arguments are shown by adding the command-line argument "--help".
```

Comment:
I think my biggest frustration right now is trying to figure out the best way to use this all from our top level makefile during our CI. I can't seem to even get cmake to actually correctly recognize the root libCEED dir

Reply:
>  top level makefile during our CI.

Could you point me to the relevant file.?

Reply:
In the top-level makefile, see the logic for `solidsexamples` or `mfemexamples`. So if `DEAL_II_DIR` is set, we need a rule to make an executable (or symlink to executable) that can be run (it gets run by `junit.py`).

Reply:
Yeah, here is the relevant parts of the toplevel makefile:

https://github.com/CEED/libCEED/blob/b698cfb9c4f7c41a310553a1468bd8185bfedcf5/Makefile#L243

https://github.com/CEED/libCEED/blob/b698cfb9c4f7c41a310553a1468bd8185bfedcf5/Makefile#L581

Ultimately, the top level makefile needs to copy the executable to `/build/dealii-bps` and then test suite will read the contents of `examples/deal.II/bps.cc` to look for a comment block like this to get the test cases

https://github.com/CEED/libCEED/blob/b698cfb9c4f7c41a310553a1468bd8185bfedcf5/examples/petsc/bps.c#L27https://github.com/CEED/libCEED/blob/b698cfb9c4f7c41a310553a1468bd8185bfedcf5/examples/petsc/bps.c#L27

I can set up the comment block, I'm mostly having trouble seeing the best way to get the top level makefile to build the deal.II example.

Our only example that uses a something other than a makefile to build currently is Nek5000, but that is a very fragile setup. The MFEM example is probably the simplest example building we have to look at?